src/main/java/org/example/learningsystem/core/security/authority/UserAuthority.java

```diff
@@ -0,0 +1,35 @@
+package org.example.learningsystem.core.security.authority;
+
+import lombok.AllArgsConstructor;
+
+/**
+ * Represents user authorities in the business domain.
+ */
+@AllArgsConstructor
+public enum UserAuthority {
```

Comment:
I recommend to rename it to `UserRole`, because authority - is what exactly we can do, but role - is group of these authorities
```
Read, Write, Manage, Administrate - authorities
ADMIN, MANAGER - roles

And ADMIN role can contain Read, Write, Manage and Administrate authorities
```

In XSUAA we are talking in context of authorities (it doesn't use concept of roles), so it means that all scopes in JWT will represent authorities only (like Administrate, Manage, TenantProvisioning)
So in code we have to do check using hasAnyAuthority('Administrate') for example